### PR TITLE
layers: Cleanup subresource adapter

### PR DESCRIPTION
**Pull request by artem-lunarg**

### CODE REVIEW

File: layers/containers/subresource_adapter.cpp [hunk 1]
```diff
@@ -166,7 +166,6 @@ RangeEncoder::RangeEncoder(const VkImageSubresourceRange& full_range, const Aspe
       mip_size_(full_range.layerCount),
       aspect_size_(mip_size_ * full_range.levelCount),
       aspect_bits_(param->AspectBits()),
-      mask_index_function_(param->MaskToIndexFunction()),
```
Comment (artem-lunarg) [inline layers/containers/subresource_adapter.cpp:169]: Not used

File: layers/containers/subresource_adapter.cpp [hunk 2]
```diff
@@ -306,7 +305,7 @@ ImageRangeEncoder::ImageRangeEncoder(const vvl::Image& image, const AspectParame
     linear_image_ = false;
 
     is_compressed_ = vkuFormatIsCompressed(image.create_info.format);
-    texel_extent_ = vkuFormatTexelBlockExtent(image.create_info.format);
+    texel_block_extent_ = vkuFormatTexelBlockExtent(image.create_info.format);
```
Comment (artem-lunarg) [inline layers/containers/subresource_adapter.cpp:308]: Rename. Some variables deal with texels and some with texel blocks. It's good to distinguish.

File: layers/containers/subresource_adapter.cpp [hunk 3]
```diff
@@ -360,16 +359,24 @@ IndexType ImageRangeEncoder::Encode2D(const VkSubresourceLayout& layout, uint32_
                                       const VkOffset3D& offset) const {
     assert(offset.z == 0U);
     // The address offset of the beginning of offset.x's block is:
-    // floor(offset.x * / texel_extent_.width) * texel_extent_.width * texel_extent_.height * texel_sizes_[apsect].
-    // Since offset.x must be a multiple of texel_extent_.width, we can simplify the formula by canceling out texel_extent_.width.
-    double xSize = static_cast<double>(texel_extent_.height * texel_sizes_[aspect_index]);
+    //      block_offset_x = block_index_x * block_height * texel_sizes_[apsect]
```
Comment (artem-lunarg) [inline layers/containers/subresource_adapter.cpp:362]: Add details to the comment. Even multiplication by height might look almost obvious, when I looked at this code the first time, and also did not work with compressed resources recently, it was less obvious.

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM. A change that is mostly - lines and + comments is always good

**Review by spencer-lunarg — APPROVED** [review]

**Pull request merged**